src/ballet/shred/fd_shred.h

```diff
@@ -237,10 +244,10 @@ fd_shred_variant( uchar type,
 FD_FN_PURE static inline ulong
 fd_shred_sz( fd_shred_t const * shred ) {
   uchar type = fd_shred_type( shred->variant );
-  return fd_ulong_if( 
+  return fd_ulong_if(
     type & FD_SHRED_TYPEMASK_CODE,
     FD_SHRED_MAX_SZ,
-    fd_ulong_if( ( type==FD_SHRED_TYPE_MERKLE_DATA ) | ( type==FD_SHRED_TYPE_MERKLE_DATA_CHAINED ),
+    fd_ulong_if( ( type==FD_SHRED_TYPE_MERKLE_DATA ) | ( type==FD_SHRED_TYPE_MERKLE_DATA_CHAINED ) | ( type==FD_SHRED_TYPE_MERKLE_DATA_CHAINED_RESIGNED ),
       FD_SHRED_MIN_SZ,
       shred->data.size ) ); /* Legacy data */
```

Comment:
You can just flip this `if`. It's only legacy data that is `shred->data.size`. Otherwise, it should be `FD_SHRED_MIN_SZ`.

Reply:
good plan